### ISSUE TITLE
Sqlite: ATTACH DATABASE

### ISSUE THREAD

**Issue by piula** [HeidiSQL/HeidiSQL#946]
## Steps to reproduce this issue

1. Connect to SQLite database A.sqlite
2. In a query tab, type "ATTACH DATABASE 'B.sqlite' AS b;"

## Current behavior

The "host" tab and the structure tree to the left don't show the database "b".

## Expected behavior

Both "host" tab and structure tree should show database "b". It should be able to work with in the same way as with the "main" database.

## Possible solution

Please apologize that I don't know.

## Environment

* HeidiSQL 11.0.0.5919 (didn't test 5920 as the release notes don't seem related to my question)
* Built-in sqlite.dll version 3.30.1
* Windows 8.1

**Comment by piula** [comment]
I forgot that furthermore the "host/databases" tab and the structure tree should provide the possiblity to attach and detach databases.

**Comment by ansgarbecker** [comment]
This should probably go into the session settings, into the "Database(s)" input box, which is otherwise useless for SQLite anyway.
Documentation for ATTACH: https://www.sqlite.org/lang_attach.html

**Comment by ansgarbecker** [comment]
Next build should at least show attached databases in the tree. And internal queries get a preceding "database." to run on the user selected database.
To test that, you can run any `ATTACH DATABASE ...` in a query tab, and refresh the database tree to see it.

But that's not the whole story. I want to support attaching databases via session manager.

**Comment by piula** [comment]
Wow, that's both fast and great! Working as expected. Thank you very much!

**Comment by ansgarbecker** [comment]
Yes, it's quite fun to work with SQLite. And I already expected someone asking for attaching additional databases in one session. It's too cumbersome to define one session for each database file.

After trying out things I am now working on changing the "Database filename" input box to support multiple database filenames, separated by pipe. There won't be support for custom aliases, which will be defined by their base filenames.

**Comment by ansgarbecker** [comment]
Next build supports selecting multiple database files per session manager:
![grafik](https://user-images.githubusercontent.com/7986591/77250096-d3ca1180-6c45-11ea-9b02-db731c779843.png)

You could probably test the whole stuff, and see if there are issues left. For example I am unsure if I should leave such a long string on the "Host" main tab.

**Comment by piula** [comment]
Great 👍

May I suggest that the context menu that appears when right-clicking on the database node (not the top-level, but the 2nd level that always contains "main" database) could replace "drop" with "detach", if it is an ATTACHed database.

And the "create new -> database" in the top level's context menu should offer a file selection dialog and ATTACH the selected database, regardless of whether it is a new or an existing database file.

The current "multi" selection worked fine for me, I only observed one small issue: when using "add database files", selecting a database in the file-open dialog and then "Cancel"ing the file-open dialog, the database file is appended to the database list, too. This should only happen on "OK".

**Comment by ansgarbecker** [comment]
Next build has two new menu items, for attaching a new database, and detaching an existing one:
![grafik](https://user-images.githubusercontent.com/7986591/77455457-2d733d00-6dfa-11ea-837f-7233bba23788.png)

Oh, I think I forgot the "main" database should not be detached... Or does that throw an SQL error anyway?

**Comment by ansgarbecker** [comment]
Ok, that is restricted by the library, so no need to do additional checks:

![grafik](https://user-images.githubusercontent.com/7986591/77455934-cdc96180-6dfa-11ea-8a48-0006ae2041c0.png)

Anything else to do here?

**Comment by ansgarbecker** [comment]
Ah, forgot to mention: that cancel issue should also be fixed.

**Comment by piula** [comment]
I will test tomorrow and report back, but at first glance, this looks good and intuitive imho!

**Comment by piula** [comment]
Is working perfectly, many thanks!
